include/ringbuffer_utils.hpp

```diff
@@ -36,6 +38,28 @@ constexpr uint64_t align_down(uint64_t x, uint64_t pow2) {
   return x & ~(pow2 - 1);
 }
 
+inline std::byte *ensure_wrap_copy_buffer(RingBuffer &rb,
+                                          size_t required_size) {
+  if (required_size == 0) {
+    return rb.wrap_copy;
+  }
+  size_t const aligned_size = align_up(required_size, kRingBufferAlignment);
```

Comment:
No need to bother about alignment, ring buffer alignment (8) will always be satisfied by memory returned by malloc / new.

Reply:
true, though code should be valid regardless of kRingBufferAlignment's value

Reply:
Better to keep things simple, static assert that kRingBufferAlignment is less than 16 and use make_unique:
```c++
inline std::byte *RingBuffer::ensure_wrap_copy_buffer(size_t required_size) {
  static_assert(kRingBufferAlignment <= 16, "kRingBufferAlignment must be <= 16");
  if (this->buffer_capacity < required_size) {
    this->buffer = std::make_unique<std::byte[]>(required_size);
    this->buffer_capacity = required_size;
  }
  return this->buffer.get();
}
```